pkg/osbuild/dnf_mark_stage.go

```diff
@@ -0,0 +1,56 @@
+package osbuild
+
+import (
+	"github.com/osbuild/images/pkg/rpmmd"
+)
+
+type DNFMarkStagePackageOptions struct {
+	Name  string `json:"name"`
+	Mark  string `json:"mark"`
+	Group string `json:"group,omitempty"`
+}
+
+type DNFMarkStageOptions struct {
+	Packages []DNFMarkStagePackageOptions `json:"packages"`
+}
+
+func (o DNFMarkStageOptions) isStageOptions() {}
+
+func (o DNFMarkStageOptions) validate() error {
+	return nil
+}
+
+func NewDNFMarkStageOptions(packages []DNFMarkStagePackageOptions) *DNFMarkStageOptions {
+	return &DNFMarkStageOptions{
+		Packages: packages,
+	}
+}
+
+type DNFMarkStage struct {
+}
+
+func NewDNFMarkStage(options *DNFMarkStageOptions) *Stage {
+	if err := options.validate(); err != nil {
+		panic(err)
+	}
+
+	return &Stage{
+		Type:    "org.osbuild.dnf.mark",
```

Comment:
The corresponding osbuild's PR introduces `dnf3` and `dnf5` stages, but no `dnf` stage, so this needs a change.

Reply:
Yea, this PR will need to be adjusted again to implement both osbuild stages and logic to select the correct one after we decided to split the osbuild stage :)